gpt4all-chat/embllm.cpp

```diff
@@ -148,6 +153,9 @@ void EmbeddingLLMWorker::requestSyncEmbedding(const QString &text)
 // this function is always called for storage into the database
 void EmbeddingLLMWorker::requestAsyncEmbedding(const QVector<EmbeddingChunk> &chunks)
 {
+    if (m_stopGenerating)
+        return;
```

Comment:
It seems like this is the more idiomatic Qt way:
```suggestion
    if (QThread::currentThread()->isInterruptionRequested())
        return;
```
And then call `m_workerThread.requestInterruption()` before `m_workerThread.wait()`. What do you think?

Reply:
I prefer the way I'm doing it with an explicit atomic of my own making.